File: Two Pointer/boj_12904_A와 B.cpp
//https://www.acmicpc.net/problem/12904
#include <iostream>
#include <string>

using namespace std;

int main() {
	ios::sync_with_stdio(false);
	cin.tie(0);
	cout.tie(0);
	string s, t;
	cin >> s >> t;

	int answer = 1;
	bool isReverse = false;

	while (s.length() != t.length()) {
		int sP = 0;
		int eP = t.length() - 1;
		if (isReverse) {
			if (t[sP] == 'B') 
				isReverse = false;
			t.erase(t.begin());
		}
		else {
			if (t[eP] == 'B')
				isReverse = true;
			t.pop_back();
			
		}
	}
	if (isReverse) {
		for (int i = s.length()-1; i >= 0; i--) {
			if (t[i] != s[s.length() - i - 1]) {
				answer = 0;
				break;
			}
		}
	}
	else {
		for (int i = 0; i < t.length(); i++) {
			if (t[i] != s[i]) {
				answer = 0;
				break;
			}
		}
	}
	cout << answer;
}

File: Two Pointer/boj_12904_A와 B.kt
//https://www.acmicpc.net/problem/12904
fun main() = with(System.out.bufferedWriter()){
    val br = System.`in`.bufferedReader()
       val S = br.readLine()
    val T = br.readLine()
    var isReverse =false
    var s=0
    var e=T.length-1
    while(e-s+1!=S.length){
        if(isReverse){
            if(T[s]=='B'){
                isReverse=false
            }
            s++
        }
        else{
            if(T[e]=='B'){
                isReverse=true
            }
            e--
        }
    }
    if(isReverse){
        for(i in 0 until S.length){
            if(S[i]!=T[e-i]){
                write("0")
                close()
                return
            }
        }
    }
    else{
        for(i in 0 until S.length){
            if(S[i]!=T[s+i]){
                write("0")
                close()
                return
            }
        }
    }
    write("1")
    close()
}
